pkg/proxy/proxy.go

```diff
@@ -63,8 +63,7 @@ func getLoadTestType(r *http.Request) apisLoadTestV1.LoadTestType {
 func (p *Proxy) Create(w http.ResponseWriter, r *http.Request) {
 	logger := mPkg.GetLogger(r.Context())
 
-	ctx, cancel := context.WithTimeout(context.Background(), loadtest.KubeTimeout)
-	defer cancel()
+	ctx := r.Context()
```

Comment:
Why remove the timeout here and for the other contexts?

I can understand doing the following but removing this deadline seems odd to me,
```suggestion
	ctx, cancel := context.WithTimeout(r.Context(), loadtest.KubeTimeout)
```

Reply:
Yeah, because I've passed this deadlines through `net/http` package. Much clean and easier and performant to be configured from first routine.

Reply:
Actually these timeout do slightly different things, since `request != client call` by making them the same you can actually get an cancelled request instead if a nice error response.

Reply:
That cancel timeout needed to cancel requests to Kube API, not request itself.

Reply:
These are not k8s call but normal API requests from our local or CI.

Reply:
It's the same and with the requests, it's much better. Because currently if the user canceled his requests, in the background, our library is trying to finish the process and unnecessarily uses resources. But with net/http context, if something happens to requests, all the context will be done and stop to trying on the background. I'm not sure if it's designed purposefully to say "yeah, even if something happens, try to finish operations"

Reply:
I agree that using the request context is better. 
I do not agree that removing the timeout for `KubeTimeout` is good. This is why I proposed using `ctx, cancel := context.WithTimeout(r.Context(), loadtest.KubeTimeout)`.
This way we allow the request cancellation to propagate and we allow k8s to timeout and still return a nice response.

Reply:
Ah sorry then I missed that! Sure, then I can change it like that